### ISSUE TITLE
Parsing expresion causes crash

### ISSUE THREAD

**Issue by munybt** [badeball/xpath-analyzer#1]
If I run this code on RunKit I get an error "Exited unexpectedly with code: 137"
```
var XPathAnalyzer = require("xpath-analyzer").default;
try{
    var analyzer = new XPathAnalyzer("//text()[contains(.,'{{'");
    console.log(analyzer.parse());
}
catch(e){
    console.log(e);
}
```
I know the expression isn't valid, but my node process crashes every time an expression like this is parsed.

**Comment by badeball** [comment]
Hi, @munybt 

Thanks for reporting this isse. It turns out that I was a bit too lenient and trusting in one of the parser components. It's fixed with 130432d and released as v.3.0.1.

I wouldn't be surprised if you were to find other, similar instances of erroneous expressions not handled correctly. If so, don't hesitate to mention it and I'll get it fixed.

**Comment by munybt** [comment]
@badeball 
Thank you for the quick reply and availability, great work :)
